### PR TITLE
Feature: Clean htmls to make it easier for LLMs to understand HTML content

### PR DESCRIPTION
**Pull request by shubhamofbce**
Closes: #103 
I have published the code to clean html into a[ pip package](https://pypi.org/project/clean-html-for-llm/) and we are using that here to keep the code clean. It has zero dependency. It just uses regex to clean the html content so we can easily import in our code without worrying about anything else.

##Context
TLDR; clean html reduces the html size by ~10x and keeps the structure intact which helps LLMs to understand the content better.

https://drive.google.com/file/d/1TaS1h7dmgQfc1p9DQ13p7RswCmBGR8GQ/view?usp=sharing
Watch the last part (21:20) of the meeting where i explained the advantage of cleaning html before passing it to llm. We found that the html size if getting reduced by ~10x in almost all cases which removes a lot of noise and makes it easier for LLMs to understand the html content.
Also it keeps the HTML structure intact which we can use for other purpose in future.
Bonus: This also reduces the token count which reduces the cost for any paid LLM API by ~10x which is a huge number when deployed on scale.

### CODE REVIEW

**Comment by shubhamofbce** [comment]
@mbrunel and @dhuynh95 We can go ahead with this for now. Please let me know if you have any questions.

**Comment by mbrunel** [comment]
Hi!
Thanks again for your new contribution, it's does the job, but does clean_html have to be on a separate package?
If possible we would prefer to only depend on widely supported packages.
Would it be possible to integrate your function directly in lavague instead?

**Review by mbrunel — CHANGES_REQUESTED** [review]
https://github.com/lavague-ai/LaVague/pull/114#issuecomment-2056688090

**Comment by shubhamofbce** [comment]
I built that package to help other people working on web agents. I will be adding more features and options to it in future that will help people working on web agents. In that case you will have to keep on pulling the code from the package if you need them.
If you still suggest putting that function here, i can easily move that here.

**Comment by dhuynh95** [comment]
Hi @shubhamofbce,

This is great!

I think @mbrunel made a good point. I guess it might depend on how much you think it would change.
We have had dependencies issues, as maintaining compatibility is not trivial, even with pinning and well-maintained projects.

I guess we can start by just having the code you proposed directly inside LaVague, and if we see a lot of activity on your pip package, we can see what we can do to integrate.

Would this seem fair to everyone?

**Comment by shubhamofbce** [comment]
This makes sense to me. I will move the finction here and update the PR.
@dhuynh95 @mbrunel

**Comment by shubhamofbce** [comment]
@mbrunel I have made the changes, I have created a html utils file and placed the function in that file.

**Comment by shubhamofbce** [comment]
I ran a simple experiment using our Bigaction dataset to check how effective is cleaning html before passing it to LLMs.
TLDR; I found that 80% match count got increased by 80-90% and 100% match count got increased by 25-50% after cleaning html. which is 🔥 

Steps performed during experiment:
I just asked open ai to give me the exact html block to perform the action given in the dataset. I tried that once with clean html and once with full html and then matched it with ground_truth_outer_html.
I repeated this for all 50 queries in the dataset twice. You can see the full details here on this sheet.
https://docs.google.com/spreadsheets/d/1Q60FJz1JzNuQFKx-moNQWPMt20v-aOBcQJSD0ISEl5E/edit#gid=612185286
image
image

**Comment by mbrunel** [comment]
very nice! I'll test it and probably merge it tomorrow

**Comment by lyie28** [comment]
Included and merged from another branch
